### ISSUE TITLE
Parts missing in warehouse

### ISSUE THREAD

**Issue by molemanBT** [MegaMek/mekhq#3912]
### Environment
What version of MekHQ does your issue apply to? 49.18 (carried over from 49.16)
What operating system are you using?  Windows, Mac, or Linux? (windows)

A bunch of things like engines etc don't show up in the warehouse tab, so I'm not able to sell them. Case in point in the saved file, parts in use - there's a 300XL engine (1 in use, 1 in 'stored'), but in the warehouse tab, it's not there, so I can't sell it.
[Crypt Commandos30620901.cpnx.gz](https://github.com/MegaMek/mekhq/files/14757197/Crypt.Commandos30620901.cpnx.gz)

[Crypt Commandos30620901.cpnx.gz](https://github.com/MegaMek/mekhq/files/14757197/Crypt.Commandos30620901.cpnx.gz)

**Comment by IllianiCBT** [comment]
The below image illustrates the issue. The items listed as stored are not present in the Warehouse, even after all other items have been removed.

<img width="541" alt="image" src="https://github.com/MegaMek/mekhq/assets/103902653/2922e424-ee85-40e8-9fb6-2e366df31bdb">

In testing, I have found that the 'ghost items' are still usable:
<img width="470" alt="image" src="https://github.com/MegaMek/mekhq/assets/103902653/caad7e9f-2e21-4439-9b96-f9fe0c915645">

This suggests the issue isn't with the item in storage, but how it is displayed in the Warehouse.

The below file is the campaign with the Warehouse emptied (barring those items not visible in the Warehouse)
[Moleman_#3912.cpnx.gz](https://github.com/MegaMek/mekhq/files/14764741/Moleman_.3912.cpnx.gz)

**Comment by JonhyBurner** [comment]
Yeah, I meant to put up a ticket but forgot. This is probably the same issue with any duplicate entries. There are several fields associated to a part in the save file that can create different entries in the warehouse. The thing is, those entries (as far as I can tell) have no bearing in any of the part's uses later on.

Case in point:
![Too many lasers!](https://github.com/MegaMek/mekhq/assets/67933202/93211973-db27-44ca-94d0-309decb6698b)

All these line have different entries in the save file:

```
<part id="76" type="mekhq.campaign.parts.equipment.EquipmentPart">
	<id>76</id>
	<name>Medium Laser</name>
	<unitTonnage>20</unitTonnage>
	<mode>EXTRA_4</mode>
	<skillMin>1</skillMin>
	<quantity>1</quantity>
	<quality>1</quality>
	<equipmentNum>-1</equipmentNum>
	<typeName>Medium Laser</typeName>
	<size>1.0</size>
	<equipTonnage>1.0</equipTonnage>
	</part>				
<part id="1331" type="mekhq.campaign.parts.equipment.EquipmentPart">
	<id>1331</id>
	<name>Medium Laser</name>
	<unitTonnage>20</unitTonnage>
	<mode>EXTRA_4</mode>
	<skillMin>1</skillMin>
	<brandNew>false</brandNew>
	<quantity>2</quantity>
	<quality>3</quality>
	<equipmentNum>-1</equipmentNum>
	<typeName>Medium Laser</typeName>
	<size>1.0</size>
	<equipTonnage>1.0</equipTonnage>
</part>
<part id="1783" type="mekhq.campaign.parts.equipment.EquipmentPart">
	<id>1783</id>
	<name>Medium Laser</name>
	<unitTonnage>75</unitTonnage>
	<mode>EXTRA_4</mode>
	<skillMin>3</skillMin>
	<brandNew>false</brandNew>
	<quantity>2</quantity>
	<quality>3</quality>
	<equipmentNum>-1</equipmentNum>
	<typeName>Medium Laser</typeName>
	<size>1.0</size>
	<equipTonnage>1.0</equipTonnage>
</part>
<part id="11798" type="mekhq.campaign.parts.equipment.EquipmentPart">
	<id>11798</id>
	<name>Medium Laser</name>
	<unitTonnage>30</unitTonnage>
	<hits>1</hits>
	<mode>EXTRA_2</mode>
	<skillMin>1</skillMin>
	<brandNew>false</brandNew>
	<quantity>1</quantity>
	<quality>3</quality>
	<equipmentNum>-1</equipmentNum>
	<typeName>Medium Laser</typeName>
	<size>1.0</size>
	<equipTonnage>1.0</equipTonnage>
</part>	
<part id="13736" type="mekhq.campaign.parts.equipment.EquipmentPart">
	<id>13736</id>
	<name>Medium Laser</name>
	<unitTonnage>70</unitTonnage>
	<mode>NORMAL</mode>
	<skillMin>4</skillMin>
	<brandNew>false</brandNew>
	<quantity>1</quantity>
	<quality>3</quality>
	<equipmentNum>-1</equipmentNum>
	<typeName>Medium Laser</typeName>
	<size>1.0</size>
	<equipTonnage>1.0</equipTonnage>
</part>			
```

The game hapilly uses some of them, but not others.
What I think makes them different when they shouldn't are:

- the unit tonnage (relates from where they were taken - unnecessary after removing the part)
- the mode (relates to the extra time used during salvaging/repairing - unnecessary after removing/repairing the part)
- the skillMin (relates to the skill of the tech during salvaging - unnecessary after removing the part, probable reason why some don't appear for use?)

All of these should return to the default values when salvaging/repairing, default being the values the parts have when you buy them. This probably has to do with repair code, whitch is a bit of a mess (devs words, not mine), but enforcing these defaults shouldn't break the code much? Maybe?
